File: 530-minimum-absolute-difference-in-bst/minimum-absolute-difference-in-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function(root) {
    const inOrderValues = _dfs(root);
    let min = Infinity;

    for (let i = 0; i < inOrderValues.length - 1; i++) {
        const diff = Math.abs(inOrderValues[i] - inOrderValues[i+1]);
        min = Math.min(min, diff);
    }

    return min;
};

const _dfs = (node) => {
    if (!node) return [];

    const left = _dfs(node.left);
    const right = _dfs(node.right);

    return left.concat([node.val]).concat(right)
}

File: 530-minimum-absolute-difference-in-bst/minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        orderedTreeValues = inOrderTraversal(root)
        minDiff = float("inf")

        for i in range(len(orderedTreeValues) - 1):
            diff = abs(orderedTreeValues[i] - orderedTreeValues[i+1])
            minDiff = min(minDiff, diff)
            
        return minDiff

def inOrderTraversal (node):
    if not node: return []

    left = inOrderTraversal(node.left)
    right = inOrderTraversal(node.right)

    return left + [node.val] + right
